New entry in settings for cache details to disable extraction of waypoints from personal cache notes

baiti opened the pull request:
see issue #8848

moving-bits commented on main/src/cgeo/geocaching/settings/Settings.java:
```diff
@@ -194,6 +194,7 @@ private static void migrateSettings() {
             e.putInt(getKey(R.string.pref_defaultNavigationTool), prefsV0.getInt(getKey(R.string.pref_defaultNavigationTool), NavigationAppsEnum.COMPASS.id));
             e.putInt(getKey(R.string.pref_defaultNavigationTool2), prefsV0.getInt(getKey(R.string.pref_defaultNavigationTool2), NavigationAppsEnum.INTERNAL_MAP.id));
             e.putBoolean(getKey(R.string.pref_debug), prefsV0.getBoolean(getKey(R.string.pref_debug), false));
+            e.putBoolean(getKey(R.string.pref_global_wp_extraction_disable), prefsV0.getBoolean(getKey(R.string.pref_global_wp_extraction_disable), false));
```
IMHO this line is not needed, as this setting is new, so no need to migrate.

baiti replied:
I wasn't sure ;) Now I have removed it.

Lineflyer commented:
OK to test (for CI)

Lineflyer commented on main/res/values/strings.xml:
```diff
@@ -600,6 +600,8 @@
     <string name="init_archived">Exclude Archived</string>
     <string name="init_summary_archived">Exclude archived caches</string>
+    <string name="init_global_wp_extraction_disable">Waypoint Extraction</string>
+    <string name="init_summary_global_wp_extraction_disable">Disable waypoint extraction from personal cache notes</string>
```
I suggest something like:
"Globally disable waypoint extraction from personal note. It can still be enabled for individual caches by deactivating the checkbox below the personal note of the cache."

baiti replied:
> I suggest something like:
> "Globally disable waypoint extraction from personal note. It can still be enabled for individual caches by deactivating the checkbox below the personal note of the cache."

Ok, consider it done. Will go up with the next push

Lineflyer commented:
OK to test

moving-bits commented on main/src/cgeo/geocaching/models/Geocache.java:
```diff
@@ -1558,7 +1558,7 @@ public boolean addWaypointsFromNote() {
     public boolean addWaypointsFromText(@Nullable final String text, final boolean updateDb, @NonNull final String namePrefix, final boolean forceExtraction) {
         boolean changed = false;
-        if (forceExtraction || !preventWaypointsFromNote) {
+        if (forceExtraction || (!Settings.isGlobalWpExtractionDisabled() && !preventWaypointsFromNote)) {
```
Not sure if I understand this condition correctly. To my understanding we wanted to implement an opt-out system. Wouldn't we need to check this in the `Geocache` class then instead of here? Distinguish on cache-level between a manually set value (true/false), which would override the global setting, and a "not yet set" value, which would activate the global setting for this cache?

Lineflyer replied:
> To my understanding we wanted to implement an opt-out system.

Cannot judge the code, but indeed an opt-out system was the idea:
- If the setting is not active (which should be default) waypoints are extracted and only surpressed if the user activates the checkbox under the specific personal note.
- It the setting is active (set by user, no default) the checkbox should be activated by default for all caches and can be individually deactivated by the user for each cache.

baiti replied:
> > To my understanding we wanted to implement an opt-out system.
> 
> Cannot judge the code, but indeed an opt-out system was the idea:
> 
> * If the setting is not active (which should be default) waypoints are extracted and only surpressed if the user activates the checkbox under the specific personal note.

That is indeed the implemented behaviour

> * It the setting is active (set by user, no default) the checkbox should be activated by default for all caches and can be individually deactivated by the user for each cache.

Ahh ok, that is not what the change does at the moment, however, I agree that is a much better. I understand that this type of change does not go into the place I put it. I have not found out yet, where in the code I could set the checkbox you mentioned but I will investigate that tomorrow.

Once I have it done, would I do another push into the same branch?

moving-bits replied:
> Ahh ok, that is not what the change does at the moment, however, I agree that is a much better. I understand that this type of change does not go into the place I put it. I have not found out yet, where in the code I could set the checkbox you mentioned but I will investigate that tomorrow.

Database change would go into `storage/DataStore`, applying the default value to `models/Geocache` (IIRC).

> Once I have it done, would I do another push into the same branch?

Yes, for the time being. (At then end, when everything is fine, we would squash the individual commits into one commit containing all the changes necessary for this PR.)

Thanks for working on this issue! Very welcomed!

eddiemuc replied:
> > * It the setting is active (set by user, no default) the checkbox should be activated by default for all caches and can be individually deactivated by the user for each cache.
> 
> Ahh ok, that is not what the change does at the moment, however, I agree that is a much better. I understand that this type of change does not go into the place I put it. I have not found out yet, where in the code I could set the checkbox you mentioned but I will investigate that tomorrow.

Sorry for late comment..
Imho, the easiest way to achieve the desired behaviour is to set the default setting of `Geocache.preventWaypointsFromNote` to the new setting. Meaning: in `Geocache.java`, change line 176:
`    private boolean preventWaypointsFromNote = false;`

to

`    private boolean preventWaypointsFromNote = Settings.isGlobalWpExtractionDisabled();`

That way the checkbox handling is automatically "done", and also the value would be correctly overridden from Database in case the cache is already stored there with a different setting.

baiti replied:
> > > * It the setting is active (set by user, no default) the checkbox should be activated by default for all caches and can be individually deactivated by the user for each cache.
> > 
> > 
> > Ahh ok, that is not what the change does at the moment, however, I agree that is a much better. I understand that this type of change does not go into the place I put it. I have not found out yet, where in the code I could set the checkbox you mentioned but I will investigate that tomorrow.
> 
> Sorry for late comment..
> Imho, the easiest way to achieve the desired behaviour is to set the default setting of `Geocache.preventWaypointsFromNote` to the new setting. Meaning: in `Geocache.java`, change line 176:
> ` private boolean preventWaypointsFromNote = false;`
> 
> to
> 
> ` private boolean preventWaypointsFromNote = Settings.isGlobalWpExtractionDisabled();`
> 
> That way the checkbox handling is automatically "done", and also the value would be correctly overridden from Database in case the cache is already stored there with a different setting.

Hi Eddie, you wouldn't believe it, that is **exactly** the change I just made ;) I have started to test it and it works as expected and as discussed in this thread yesterday.

I am going to push it shortly. Please have a look at the code and prefereably I'd also want some of you to give it a real try and walk through some usage scenarios on the device.

baiti replied:
I have just committed it into yesterday's pull request. I hope this was the right way doing it from a process point of view.

baiti commented:
I have actually already started to do so. First I deleted my fork on github, then I deleted my local clone and started all over. I clicked on edit in the README.md and noticed that "the system" edited the README.md in my fork and that everything I do on the Wiki, happens in my fork. Now I was just wondering, when I do a PR at some point in time, will the links I put into the Wiki be automatically converted to point to cgeo/cgeo as opposed to baiti/cgeo when a mergs is done?

Narrat commented:
> > One additional note: while I too like to work almost completely "manually" and on the command line with git, resolving conflicts is the one thing where I appreciate the help of Android Studio.
> 
> Good to know, will try it the next time. Thanks for the hint!

Worth noting that on the commandline is the command mergetool for those cases. Or if someone don't want to manually search for the conflicts. The diff program can be freely configured (vimdiff, meld and whatnot)
